### DISCUSSION TITLE
GPL-2.0-only or GPL-2.0-or-later?

### DISCUSSION THREAD

**Discussion by vhelin** [vhelin/wla-dx#516] [General]
As asked in https://github.com/vhelin/wla-dx/issues/515 - any comments? I'm thinking we should switch to GPL-2.0-or-later

**Comment by cr1901** [comment]
I don't think I have an informed opinion.

I know a lot of libraries don't like GPL 3.0, but seeing as this is an application, not a library, I'm not sure complaints apply. And I'm being nebulous because I don't actually _remember_ the complaints :D.

**Comment by vhelin** [comment]
I'm also not 100% certain that GPL-2.0-or-later would be a better choice. I just have a feeling that GPL-2.0-or-later would be more permissive license, in a good way.

Yeah, WLA DX is not really a library, but it can be used in the same ~fashion as a library: included in other projects as tools, or build an app around it like I've done in https://github.com/vhelin/SameSameC which has WLA DX as a Git submodule; the compiler generates WLA DX asm and the linker "links" everything together and calls WLA to assemble (and link) the asm code. There would be no SameSameC without WLA DX. I could have just used compiled WLA DX binaries there, but maintaining them would mean more work...

**Comment by vhelin** [comment]
Ok, if there are no objections, I'm changing the license to be GPL-2.0-or-later at the end of July 2022...
